### ISSUE TITLE
--hidden + .ignore showing ignored files inside hidden directories

### ISSUE THREAD

**Issue by dan-santana** [sharkdp/fd#237]
This bug is similar to https://github.com/BurntSushi/ripgrep/issues/807: `fd --hidden '' .` is showing an  ignored file inside a hidden directory. The following script can reproduce the issue:

```sh
mkdir -p .a/b .a/c
touch .a/b/file .a/c/file
echo .a/b > .ignore
ln -rsf .ignore .gitignore
fd --hidden '' .
```

The output will be:
```
.a
.a/b
.a/b/file
.a/c
.a/c/file
.ignore
```

The expected output is:

```
.a
.a/c
.a/c/file
.ignore
```

**As noted in the ripgrep issue, it happens only with `fd --hidden '' .`, but not with `fd --hidden '' ./`.**

**Comment by sharkdp** [comment]
Thank you for reporting this. I'll keep this open to track the upstream issue.

**Comment by sharkdp** [comment]
Closed by updating to ignore-0.4.1 in 8805bd32e284e6e6d0d37f0041a002a1d90adcbf

```
▶ fd --hidden
.a
.a/b
.a/b/file
.a/c
.a/c/file
.gitignore
.ignore
```
